### ISSUE TITLE
Array field that includes a gallery with hasMany set to true crashes the app

### ISSUE THREAD

**Issue by bmamouri** [payloadcms/payload#7857]
### Describe the Bug

Having an upload filed with hasMany set to true in an array field causes the app to crash.

Please see the collection below:

```js
export const Product: CollectionConfig = {
  slug: 'product',
  admin: {
    group: 'Products',
    useAsTitle: 'title',
  },
  fields: [
    {
      name: 'title',
      type: 'text',
      required: true,
    },
    {
      name: 'variants',
      type: 'array',
      fields: [
        {
          name: 'code',
          type: 'text',
        },
        {
          name: 'name',
          type: 'text',
        },
        {
          name: 'gallery',
          type: 'upload',
          relationTo: 'media',
          hasMany: true,
          required: true,
          maxDepth: 2,
        },
      ],
    },
  ],
}
```

This is how it looks in the admin app:
<img width="491" alt="image" src="https://github.com/user-attachments/assets/365a1a3b-e1bf-404e-b9bc-ae00efa42336">

But as soon as I click "Add Variant" the app crashes with the following error:
<img width="559" alt="image" src="https://github.com/user-attachments/assets/6f5aa99b-7011-4648-ac4e-1dd3257b52d0">

Interestingly, if I add an empty record in the database for the "variant" field, I can add images with no issue:

<img width="491" alt="image" src="https://github.com/user-attachments/assets/99a6b5b4-b12b-4774-8b85-e01dd4034506">

### To Reproduce

I have a minimal repo to show the issue: https://github.com/bmamouri/payload-bug

### Payload Version

3.0.0-beta.91

**Comment by bmamouri** [comment]
I have just reliased that this has nothing to do with array field. Even if the field does is not in array and is inside another collection this will happen:

Products
```js
export const Product: CollectionConfig = {
  slug: 'product',
  fields: [
    {
      name: 'title',
      type: 'text',
      required: true,
    },
    {
      name: 'variants',
      type: 'relationship',
      relationTo: 'variant' as CollectionSlug,
      hasMany: true,
    },
  ],
}
```

Variants
```js
import { CollectionConfig, CollectionSlug } from 'payload'

export const Variant: CollectionConfig = {
  slug: 'variant',
  fields: [
    {
      name: 'code',
      type: 'text',
    },
    {
      name: 'name',
      type: 'text',
    },
    {
      name: 'gallery',
      type: 'upload',
      relationTo: 'media',
      hasMany: true, // <= Adding this line will crash the app
    },
  ],
}
```

**Comment by bmamouri** [comment]
I am closing as this issue is fixed in 3.0.0-beta.95. Thanks Payload CMS team. 🙏

**Comment by ivan-kardash** [comment]
I've noticed that the bug is reproducing if you have **Global** that includes a field with type 'array', that consists of a field with type 'relation' that refers to collection.
The error:
```
ERROR: Error: cannot connect to Postgres. Details: Cannot read properties of undefined (reading 'Symbol(drizzle:Name)')
```
My simplified config:
```ts
const Navigation: GlobalConfig = {
  slug: 'navigation',
  fields: [
    {
      name: 'areas',
      type: 'array',
      fields: [
        {
          name: 'nav',
          type: 'relationship',
          relationTo: 'navigationItems', // my collection slug
          localized: true,
          hasMany: true,
        },
      ],
    },
  ],
};
```
The versions of packages:
```
 "payload": "3.0.0-beta.98",
```

Could this issue be reopened?
